### DISCUSSION TITLE
I would like to ask a question about thread concurrency

### DISCUSSION THREAD

**Discussion by xiaoxiongxyy** [vectordotdev/vector#19757] [Q&A]
We want to use Vector to collect Kubenetes logs, but there are 500 pods on a node. To use vector demonset to collect vector, do we need to start independent threads to collect 500 files? Can such a high thread count be supported? I want to know the detailed threading model of this piece. I hope I can get a reply. Thank you. Oh, by the way, we have replaced logstash.
Vector feels great

**Comment by jszwedko** [comment]
Hi @xiaoxiongxyy ! If I remember correctly, the `file` source uses a single thread for reading.

**Reply by xiaoxiongxyy** [reply]
1 file source 1 thread or 100 file source 1 thread？please

**Reply by jszwedko** [reply]
1 `file` source, 1 reading thread. If you had multiple `file` sources, they would each use their own.
